src/pages/workspace/WorkspacesListPage.tsx

```diff
@@ -442,7 +456,13 @@ function WorkspacesListPage() {
         [shouldUseNarrowLayout],
     );
 
-    const navigateToDomain = useCallback(() => openOldDotLink(CONST.OLDDOT_URLS.ADMIN_DOMAINS_URL), []);
+    const navigateToDomain = useCallback(({accountID, isValidated}: {accountID: number; isAdmin: boolean; isValidated: boolean}) => {
+        if (isValidated) {
+            openOldDotLink(CONST.OLDDOT_URLS.ADMIN_DOMAINS_URL);
+        } else {
+            Navigation.navigate(ROUTES.WORKSPACES_VERIFY_DOMAIN.getRoute(accountID));
+        }
```

Comment:
We prefer early returns in such functions:

```suggestion
        if (isValidated) {
            return openOldDotLink(CONST.OLDDOT_URLS.ADMIN_DOMAINS_URL);
        } 
        
        Navigation.navigate(ROUTES.WORKSPACES_VERIFY_DOMAIN.getRoute(accountID));
```